core/shared/src/main/scala/org/http4s/UrlForm.scala

```diff
@@ -95,11 +95,25 @@ object UrlForm {
     if (values.get("").fold(false)(_.isEmpty)) new UrlForm(values - "")
     else new UrlForm(values)
 
+  def single(key: String, value: String): UrlForm =
+    new UrlForm(Map(key -> Chain.one(value)))
+
   def apply(values: (String, String)*): UrlForm =
-    values.foldLeft(empty)(_ + _)
+    values match {
+      case Seq() => empty
+      case Seq(x) => single(x._1, x._2)
+      case h +: tail => tail.foldLeft(single(h._1, h._2))(_ + _)
+    }
 
   def fromChain(values: Chain[(String, String)]): UrlForm =
-    apply(values.toList: _*)
+    values.knownSize match {
+      case 0 => empty
+      case 1 =>
+        val h = values.headOption.get
```

Comment:
Perhaps, an intermediate `Option` is not necessary here:
```suggestion
        val h = values.head
```
if I am not missing something...

Reply:
You're absolutely right!

Reply:
Ah, the compiler just reminded me, why I used here `headOption`:
```
value head is not a member of cats.data.Chain[(String, String)]
```